### PR TITLE
Rewrite GoldenRatio in terms of sqrt

### PR DESCRIPTION
**Pull request by nc1337**
Now the ouput is 

```
In [2]: GoldenRatio.rewrite(sqrt)
Out[2]:
      ___
1   \/ 5
- + -----
2     2
```

Also added tests.

---

TODO
- [x] should this be written as `_eval_expand_func`?

### CODE REVIEW

**Comment by Nitin216** [comment]
Someone review this PR.

**Comment by smichr** [comment]
+1 if tests pass

File: sympy/core/tests/test_numbers.py [hunk 1]
```diff
@@ -1544,3 +1544,6 @@ def test_mod_inverse():
     raises(TypeError, lambda: mod_inverse(2, x))
     raises(ValueError, lambda: mod_inverse(2, S.Half))
     raises(ValueError, lambda: mod_inverse(2, cos(1)**2 + sin(1)**2))
+
```
Comment (leosartaj) [inline sympy/core/tests/test_numbers.py]: Add a newline here.

File: sympy/core/numbers.py [hunk 2]
```diff
@@ -3311,6 +3311,10 @@ def _sage_(self):
         import sage.all as sage
         return sage.golden_ratio
 
+    def _eval_rewrite_as_sqrt(self):
+        from sympy import sqrt
+        return S.Half + S.Half*sqrt(5)
```
Comment (leosartaj) [inline sympy/core/numbers.py]: Can this be simply ``_eval_rewrite_as_sqrt = _eval_expand_func``?

**Comment by moorepants** [comment]
Please use more informative pull request titles in the future.

**Comment by Nitin216** [comment]
@moorepants Sorry, will do next time.

**Comment by Nitin216** [comment]
Could someone merge this PR.

**Comment by leosartaj** [comment]
@Nitin216 Could you please address my [comment](https://github.com/sympy/sympy/pull/10900#discussion_r57342825).

File: sympy/core/numbers.py (hunk 2 above)
Comment (Nitin216) [inline sympy/core/numbers.py]: @leosartaj In GoldenRatio, there is already an ```_eval_expand_func```. So what do u want

Comment (leosartaj) [inline sympy/core/numbers.py]: You can replace the above lines of code with just ``_eval_rewrite_as_sqrt = _eval_expand_func``.

Comment (Nitin216) [inline sympy/core/numbers.py]: @leosartaj Got that. Thank you

**Comment by leosartaj** [comment]
+1

**Comment by leosartaj** [comment]
@Nitin216 By writing something like 'fixes #10894' in the PR description, closes the issue automatically when the PR is merged.

**Comment by leosartaj** [comment]
Thanks! @Nitin216. This is in.

**Pull request merged**